integration_test_suite/integration_tests/pom.xml

```diff
@@ -0,0 +1,231 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.prometheus.jmx</groupId>
+        <artifactId>integration_test_suite</artifactId>
+        <version>0.18.1-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>integration_tests</artifactId>
+    <name>Prometheus JMX Exporter - Integration Test Suite</name>
+    <description>
+        See https://github.com/prometheus/jmx_exporter/blob/master/README.md
+    </description>
+    <url>http://github.com/prometheus/jmx_exporter</url>
+
+    <licenses>
+        <license>
+            <name>The Apache Software License, Version 2.0</name>
+            <url>http://www.apache.org/licenses/LICENSE-2.0.txt</url>
+            <distribution>repo</distribution>
+        </license>
+    </licenses>
+
+    <properties>
+        <maven.compiler.source>11</maven.compiler.source>
+        <maven.compiler.target>11</maven.compiler.target>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <jmx.core.root>../..</jmx.core.root>
+        <!-- smoke test containers -->
+        <docker.image.names>
+            azul/zulu-openjdk:8
+            azul/zulu-openjdk:11
+            azul/zulu-openjdk:17
+            azul/zulu-openjdk:19
+            azul/zulu-openjdk:20
+            openjdk:6
+            ticketfly/java:6
+        </docker.image.names>
+        <!-- all test containers (empty value) -->
+        <!-- <docker.image.names/> -->
+    </properties>
+
+    <repositories>
+        <repository>
+            <id>test-engine</id>
+            <url>https://repository.antublue.org/test-engine</url>
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+            <snapshots>
+                <enabled>true</enabled>
+            </snapshots>
+        </repository>
+    </repositories>
+
+    <pluginRepositories>
+        <pluginRepository>
+            <id>test-engine-maven-plugin</id>
+            <url>https://repository.antublue.org/test-engine-maven-plugin</url>
+            <releases>
+                <enabled>true</enabled>
+            </releases>
+            <snapshots>
+                <enabled>true</enabled>
+            </snapshots>
+        </pluginRepository>
+    </pluginRepositories>
```

Comment:
Hi Doug, I failed to run `./mvnw clean verify` locally, because the antublue dependencies cannot be retrieved:
```
Plugin org.antublue:test-engine-maven-plugin:0.0.3 or one of its dependencies could not be resolved
```

I'm wondering if the plugin is really needed? Is it used for parameterized tests? The current integration tests use  `Parameterized`, which is standard JUnit 5 functionality that does not require other dependencies.

https://github.com/prometheus/jmx_exporter/blob/9ad31dac2d0c6d49d746efd6f5380dab6c78850b/integration_tests/smoke_tests/src/test/java/io/prometheus/jmx/AgentJavaVersionsIT.java#L26

What do you think about using JUnit instead of Antublue?

Reply:
Hi Fabian, can you please re-test using `./mvnw -U clean verify` (... there was a CloudFlare/repository issue.)

`Parameterized` and `@RunWith` included with JUnit 5 are really JUnit 4. JUnit 5 has a feature gap testing parameterized test classes: https://github.com/junit-team/junit5/issues/878

I can refactor to use JUnit 4, but I feel the AntuBLUE Test Engine provides a lot of benefits...

- better console test output
- parallel execution by default (console & IntelliJ)
- test class include/exclude filters
- test method include/exclude filters
- test class tag include/exclude filters

... all of which make development/testing much easier.

Reply:
Example AntuBLUE Test Engine output...

```
...
2023-03-28 | 12:49:58.101 | test-engine-08 | INFO | ticketfly/java:6 | Container ticketfly/java:6 started in PT0.387857S
2023-03-28 | 12:50:05.108 | test-engine-08 | INFO | ticketfly/java:6 | (TEST) io.prometheus.jmx.test.httpserver.general.StartDelaySeconds_IT testHealthy()
2023-03-28 | 12:50:05.110 | test-engine-08 | INFO | ticketfly/java:6 | (PASS) io.prometheus.jmx.test.httpserver.general.StartDelaySeconds_IT testHealthy()
2023-03-28 | 12:50:05.110 | test-engine-08 | INFO | ticketfly/java:6 | (TEST) io.prometheus.jmx.test.httpserver.general.StartDelaySeconds_IT testMetrics()
2023-03-28 | 12:50:05.288 | test-engine-08 | INFO | ticketfly/java:6 | (PASS) io.prometheus.jmx.test.httpserver.general.StartDelaySeconds_IT testMetrics()
2023-03-28 | 12:50:05.288 | test-engine-08 | INFO | ticketfly/java:6 | (TEST) io.prometheus.jmx.test.httpserver.general.StartDelaySeconds_IT testMetricsOpenMetricsFormat()
2023-03-28 | 12:50:05.349 | test-engine-08 | INFO | ticketfly/java:6 | (PASS) io.prometheus.jmx.test.httpserver.general.StartDelaySeconds_IT testMetricsOpenMetricsFormat()
2023-03-28 | 12:50:05.349 | test-engine-08 | INFO | ticketfly/java:6 | (TEST) io.prometheus.jmx.test.httpserver.general.StartDelaySeconds_IT testMetricsPrometheusFormat()
2023-03-28 | 12:50:05.398 | test-engine-08 | INFO | ticketfly/java:6 | (PASS) io.prometheus.jmx.test.httpserver.general.StartDelaySeconds_IT testMetricsPrometheusFormat()
2023-03-28 | 12:50:06.191 | test-engine-08 | INFO | ticketfly/java:6 | (PASS) io.prometheus.jmx.test.httpserver.general.StartDelaySeconds_IT
2023-03-28 | 12:50:06.574 | main | INFO | -----------------------------------
2023-03-28 | 12:50:06.574 | main | INFO | AntuBLUE Test Engine v2.0.5 Summary
2023-03-28 | 12:50:06.574 | main | INFO | -----------------------------------
2023-03-28 | 12:50:06.574 | main | INFO | 
2023-03-28 | 12:50:06.574 | main | INFO | TESTS : 595, PASSED : 595, FAILED : 0, SKIPPED : 0
2023-03-28 | 12:50:06.574 | main | INFO | 
2023-03-28 | 12:50:06.574 | main | INFO | -----------------------------------
2023-03-28 | 12:50:06.574 | main | INFO | PASSED
2023-03-28 | 12:50:06.574 | main | INFO | -----------------------------------
2023-03-28 | 12:50:06.574 | main | INFO | Total Time  : 1 minute, 34 seconds, 979 ms
2023-03-28 | 12:50:06.576 | main | INFO | Finished At : 2023-03-28T12:50:06.576-04:00
2023-03-28 | 12:50:06.576 | main | INFO | -----------------------------------
```